app/services/cache_service.rb

```diff
@@ -43,6 +46,13 @@ def listing_unchanged?(prev_cached_listing, fresh_listing)
       (prev_cached_listing['LastModifiedDate'] == fresh_listing['LastModifiedDate'])
   end
 
+  def listing_images_unchanged?(prev_cached_listing, fresh_listing)
```

Comment:
Need to ensure that `==` check is sufficient or if we need to manually check attributes